app/models/test_run.rb

```diff
@@ -86,16 +55,10 @@ def build_step_info(step_status)
     step_info
   end
 
-  def calc_live_times(steps_array, step_info)
-    step_info[:pending_time] = "n/a"
-    step_info[:soaking_time] = "n/a"
-    step_info[:run_time] = step_info[:duration]
-  end
-
   def calc_times(steps_array, step_info)
     temp = {}
     steps_array.each_with_object(temp) {|s, memo| memo[s.status] = s.started_at }
-    temp["pending"] ? step_info[:pending_time] = (temp["soaking"] - temp["pending"]) : step_info[:pending_time] = 0
+    temp["pending"] && temp["soaking"] ? step_info[:pending_time] = (temp["soaking"] - temp["pending"]) : step_info[:pending_time] = 0
```

Comment:
this is really dense.  I would like to see this paragraph written with much shorter sentences.